docs/develop/dotnet/redis-om-dotnet/aggregations/intro/intro.md

```diff
@@ -0,0 +1,84 @@
+
+```csharp
+[Document]
+public class Employee
+{
+    [Indexed]
+    public string Name { get; set; }
+    
+    [Indexed]
+    public GeoLoc? HomeLoc { get; set; }
+
+    [Indexed(Aggregatable = true)]
+    public int Age { get; set; }
+
+    [Indexed(Aggregatable = true)]
+    public double Sales { get; set; }
+    
+    [Indexed(Aggregatable = true)]
+    public double SalesAdjustment { get; set; }
+
+    [Searchable(Aggregatable = true)]
+    public string Department { get; set; }
+}
+```
+
+We'll then create the index for that model, pull out a `RedisAggregationSet<T>` from our provider, and initialize the index, and seed some data into our database
+
+```csharp
+var provider = new RedisConnectionProvider("redis://localhost:6379");
+await provider.Connection.CreateIndexAsync(typeof(Restaurant));
+var employees = provider.RedisCollection<Employee>();
+var employeeAggregations = provider.AggregationSet<Employee>();
+var e1 = new Employee {Name = "Bob", Age = 35, Sales = 100000, SalesAdjustment = 1.5,  Department = "EMEA Sales"};
+var e2 = new Employee {Name = "Alice", Age = 52, Sales = 300000, SalesAdjustment = 1.02, Department = "Partner Sales"};
+var e3 = new Employee {Name = "Marcus", Age = 42, Sales = 250000, SalesAdjustment = 1.1, Department = "NA Sales"};
+var e4 = new Employee {Name = "Susan", Age = 27, Sales = 200000, SalesAdjustment = .95, Department = "EMEA Sales"};
+var e5 = new Employee {Name = "John", Age = 38, Sales = 275000, SalesAdjustment = .9, Department = "APAC Sales"};
+var e6 = new Employee {Name = "Theresa", Age = 30, Department = "EMEA Ops"};
+employees.Insert(e1);
+employees.Insert(e2);
+employees.Insert(e3);
+employees.Insert(e4);
+employees.Insert(e5);
+employees.Insert(e6);
+```
+
+## The AggregationResult
+
+The Aggregations pipeline Is all built around the `RedisAggregationSet<T>` this Set is generic, so you can provide the model that you want to build your aggregations around (an Indexed type), but you will notice that the return type from queries to the `RedisAggregationSet` is the generic type passed into it. Rather it is an `AggregationResult<T>` where `T` is the generic type you passed into it. This is a really important concept, when results are returned from aggregations, they are not hydrated into an object like they are with queries. That's because Aggregations aren't meant to pull out your model data from the database, rather they are meant to pull out aggregated results. The AggregationResult has a `RecordShell` field, which is ALWAYS null outside of the pipeline. It can be used to build expressions for querying objects in Redis, but when the AggregationResult lands, it will not contain a hydrated record, rather it will contain a dictionary of Aggregations built by the Aggregation pipeline. This means that you can access the results of your aggregations by indexing into the AggregationResult.
```

Comment:
```suggestion
The Aggregations pipeline is all built around the `RedisAggregationSet<T>` this Set is generic, so you can provide the model that you want to build your aggregations around (an Indexed type), but you will notice that the return type from queries to the `RedisAggregationSet` is the generic type passed into it. Rather it is an `AggregationResult<T>` where `T` is the generic type you passed into it. This is a really important concept, when results are returned from aggregations, they are not hydrated into an object like they are with queries. That's because Aggregations aren't meant to pull out your model data from the database, rather they are meant to pull out aggregated results. The AggregationResult has a `RecordShell` field, which is ALWAYS null outside of the pipeline. It can be used to build expressions for querying objects in Redis, but when the AggregationResult lands, it will not contain a hydrated record, rather it will contain a dictionary of Aggregations built by the Aggregation pipeline. This means that you can access the results of your aggregations by indexing into the AggregationResult.
```